src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -1282,6 +1308,59 @@ bool CommandLineInterface::GenerateOutput(
   return true;
 }
 
+bool CommandLineInterface::GenerateDependencyManifestFile(
+    const vector<const FileDescriptor*> parsed_files,
+    DiskSourceTree * source_tree) {
+  FileDescriptorSet file_set;
+
+  set<const FileDescriptor*> already_seen;
+  for (int i = 0; i < parsed_files.size(); i++) {
+    GetTransitiveDependencies(parsed_files[i],
+                              false,
+                              &already_seen, file_set.mutable_file());
+  }
+
+  int fd;
+  do {
+    fd = open(dependency_manifest_name_.c_str(),
```

Comment:
open is in cpp standard library. It's safe to use it cross platform.
I agree this doesn't look good. But now FileOutputStream is supposed to be used in this way. It requires fd, which can only be gotten like this. Other place in this file has same code. Will fix it internally.